client/src/modules/user/model/actions.js

```diff
@@ -107,3 +116,21 @@ export const updatePassword = (token, password, passwordConfirmation) =>
 
 export const resendRecoveryLink = token =>
   postData(`/auth/resend-recovery-link/${token}`);
+
+export const fetchUserDetails = userName => dispatch =>
+  getData('/auth/user-details')
```

Comment:
What about `getJson = (...args) => getData(...args).then(response => response.json())` to simplify many of the calls which expect JSON output?